src/System.Console/src/System/IO/SyncTextReader.cs

```diff
@@ -31,8 +31,8 @@ protected override void Dispose(bool disposing)
             {
                 lock (_methodLock)
                 {
-                    // Explicitly pick up a potentially methodimpl'ed Dispose
-                    ((IDisposable)_in).Dispose();
+                    // This will always pick up TextReader.Dispose
```

Comment:
I don't know the history behind this particular code, but think about the time it was written:

TextReader only had Close and explicitly implemented Dispose. Close was virtual and there was a virtual Dispose(bool). Perhaps they were attempting to skip some overriden Close behavior?